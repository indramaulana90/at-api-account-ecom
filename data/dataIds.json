[
	{
		"relativePath": "__tests__/v1/example/post/example/data.js",
		"ids": {
			"idUsers": 999999999
		}
	},
	{
		"relativePath": "__tests__/v1/order/get/order_parcel_detail/data.js",
		"ids": {
			"idOrders": 839640065,
			"idOrderParcels": 839641572,
			"idOrderParcelProducts": 839644779,
			"idBrands": "888888f39970108f39970108",
			"idBuyers": 839966387,
			"idUrbans": 839966387,
			"idInvoiceGroups": "888888f39970108f39970108",
			"idCart": "888888f39970314f39970314",
			"idDevice": "839966387",
			"idProducts": "888888f39970108f39970108",
			"idPhoneNumber": 839981005
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/get/workday_list/data.js",
		"ids": {
			"idWorkingDays": [
				841736409,
				841736511,
				841736535
			]
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/get/workday_detail/data.js",
		"ids": {
			"idWorkingDays": [
				841802581,
				841802625
			]
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/delete/workingDay/data.js",
		"ids": {
			"idWorkingDays": [
				841888623,
				841888635,
				841888672
			]
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/patch/update_workingday/data.js",
		"ids": {
			"idWorkingDays": [
				841909018,
				841909045,
				841909072
			]
		}
	},
	{
		"relativePath": "__tests__/v1/order/post/cancel_order_parcel/data.js",
		"ids": {
			"idOrders": 842063620,
			"idOrderParcels": 842063620,
			"idOrderParcelProducts": 842063620,
			"idBrands": "888888f42063620f42063620",
			"idBuyers": 842063620,
			"idUrbans": 842063620,
			"idInvoiceGroups": "888888f42063620f42063620",
			"idCart": "888888f42063620f42063620",
			"idDevice": 842063620,
			"idProducts": "888888f42063620f42063620",
			"idPhoneNumber": 842063620
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/get/holiday_list/data.js",
		"ids": {
			"idHolidays": [
				841986121,
				841986145,
				841986166,
				841986181,
				841986198
			]
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/get/holiday_detail/data.js",
		"ids": {
			"idHolidays": [
				841994725,
				841994752
			]
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/delete/holidays/data.js",
		"ids": {
			"idHolidays": [
				841999606,
				841999633,
				841999652
			]
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/post/holiday/data.js",
		"ids": {
			"idHolidays": 842147917
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/patch/holiday_update/data.js",
		"ids": {
			"idHolidays": [
				842318652,
				842318674,
				842318692,
				842321145
			]
		}
	},
	{
		"relativePath": "__tests__/v1/order/patch/update_order_status/data.js",
		"ids": {
			"idOrders": 842412325,
			"idOrderParcels": 842412325,
			"idOrderParcelProducts": 842412325,
			"idBrands": "888888f42412325f42412325",
			"idBuyers": 842412325,
			"idUrbans": 842412325,
			"idInvoiceGroups": "888888f42412325f42412325",
			"idCart": "888888f42412325f42412325",
			"idDevice": 842412325,
			"idProducts": "888888f42412325f42412325",
			"idPhoneNumber": 842412325
		}
	},
	{
		"relativePath": "__tests__/v1/buyer_order/patch/order_sellers_is_read/data.js",
		"ids": {
			"idOrders": 856578262,
			"idOrderSellers": 856578315
		}
	},
	{
		"relativePath": "__tests__/v1/buyer_order/patch/update_order_seller_failed/data.js",
		"ids": {
			"idOrders": 856642262,
			"idOrderSellers": 856642262,
			"idCart": "888888f56642550f56642550",
			"idDevice": 856642550,
			"idOrderBilling": 856642736,
			"idOrderSellerStatusHistoriesId": 856642736,
			"idOrderProduct": 856642736
		}
	},
	{
		"relativePath": "__tests__/v1/buyer_order/patch/update_order_seller_cancel/data.js",
		"ids": {
			"idOrders": 856664266,
			"idOrderSellers": 856664266
		}
	},
	{
		"relativePath": "__tests__/v1/warehouse/post/assign_product/data.js",
		"ids": {
			"idWarehouses": 862452416,
			"idSyncProduct": "62ec8be49c912a4015db7b41"
		}
	},
	{
		"relativePath": "__tests__/v1/warehouse/patch/warehouse_assign_product/data.js",
		"ids": {
			"idSyncProduct": "62ec8be49c912a4015db7b41"
		}
	}
]