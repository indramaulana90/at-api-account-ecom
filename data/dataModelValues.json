{
    "banners":{
        "name": "ATAPI ${params.atName} Banner Name",
        "header": "ATAPI${params.atName} Header",
        "description": "ATAPI ${params.atName} Description",
        "termAndCondition": "<p>• Berlaku untuk user dari semua tier</p><p>• Pembelian min 1.000.000 untuk SKU (SGM 300gr)</p><p>• Tidak berlaku kelipatan</p>"
    },
    "brands": {
        "name": "ATAPI ${params.atName} Brand Name",
        "code": "ATAPI${params.atName}Brand"
    },
    "buyers": {
        "name": "ATAPI ${params.atName} Buyers Name",
        "address": "ATAPI ${params.atName} Address"
    },
    "catalogues": {
        "name": "ATAPI ${params.atName} Catalogue Name",
        "code": "ATAPI${params.atName}Cata",
        "detail": "ATAPI ${params.atName} Catalogue Detail",
        "description": "ATAPI ${params.atName} Catalogue Description",
        "information": "ATAPI ${params.atName} Catalogue Information"
    },
    "catalogueCategories": {
        "name": "ATAPI ${params.atName} Category Name",
        "code": "ATAPI${params.atName}CCat"
    },
    "catalogueUnits": {
        "name": "ATAPI ${params.atName} Catalogue Unit Name"
    },
    "channels": {
        "name": "ATAPI ${params.atName} Channel Name",
        "externalId": "ATAPI${params.atName}Chan",
        "description": "ATAPI ${params.atName} Description"
    },
    "clusters": {
        "name": "ATAPI ${params.atName} Cluster Name",
        "externalId": "ATAPI${params.atName}Clus",
        "description": "ATAPI ${params.atName} Description"
    },
    "groups": {
        "name": "ATAPI ${params.atName} Group Name",
        "externalId": "ATAPI${params.atName}Grup",
        "description": "ATAPI ${params.atName} Description"
    },
    "invoiceGroups": {
        "name": "ATAPI ${params.atName} Invoice Group name",
        "code": "ATAPI${params.atName}InGr",
        "externalId": "ATAPI${params.atName}InGr"
    },
    "mssCatalogues": {
        "name": "ATAPI ${params.atName} MssCatalogues Name",
        "externalId": "ATAPI${params.atName}MSS"
    },
    "portfolios": {
        "name": "ATAPI ${params.atName} Portfolio Name",
        "externalId": "ATAPI${params.atName}Ptfl"
    },
    "products": {
        "name": "ATAPI ${params.atName} Product Name",
        "externalId": "ATAPI${params.atName}Pdct"
    },
    "stores": {
        "address": "ATAPI ${params.atName} Stores Address",
        "name": "ATAPI ${params.atName} Stores Name",
        "noteAddress": "ATAPI ${params.atName} Stores Note Address",
        "externalId": "ATAPI${params.atName}Strs"
    },
    "salesRep": {
        "name": "ATAPI ${params.atName} Users Sales fullname",
        "externalId": "ATAPI${params.atName}UsSl"
    },
    "salesTeamStoreChannels": {
        "name": "ATAPI ${params.atName} ST Store Channels Name"
    },
    "types": {
        "name": "ATAPI ${params.atName} Type Name",
        "externalId": "ATAPI${params.atName}Type",
        "description": "ATAPI ${params.atName} Description"
    },
    "urbans": {
        "city": "ATAPI ${params.atName} City",
        "district": "ATAPI ${params.atName} District",
        "urban": "ATAPI ${params.atName} Urban"
    },
    "users": {
        "fullname": "ATAPI ${params.atName} users fullname",
        "email":"ATAPI${params.atName}@atapi.com",
        "userCode": "ATAPI${params.atName}Usrs"
    },
    "usersSales": {
        "fullname": "ATAPI ${params.atName} users sales fullname",
        "email":"ATAPI${params.atName}@sales.com",
        "userCode": "ATAPI${params.atName}UsSl"
    },
    "vouchers": {
        "name": "ATAPI {params.atName} Name",
        "externalId": "ATAPI${params.atName}Vouc",
        "code": "${params.atName}",
        "description": "ATAPI ${params.atName} Description",
        "header":"ATAPI ${params.atName} Header",
        "terms": "{ATAPI ${params.atName} Terms and Condition}",
        "instructions": "{ATAPI ${params.atName} Instructions}",
        "tag": "{ATAPI ${params.atName} Tag}"
    },
    "warehouses": {
        "name": "ATAPI ${params.atName} Warehouse name",
        "code": "ATAPI${params.atName}WH",
        "externalId": "ATAPI${params.atName}WH",
        "noteAddress": "ATAPI ${params.atName} Note Address",
        "address": "ATAPI ${params.atName} Address"
    },
    "warehouseValues": {
        "name": "ATAPI ${params.atName} Warehouse Value name"
    },
    "warehouseTemperatures": {
        "name": "ATAPI ${params.atName} Warehouse Temperature name"
    }
}