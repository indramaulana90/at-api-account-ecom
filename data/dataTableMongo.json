[
	{
		"idInvoiceGroups": {
			"isMongo": true
		}
	},
	{
		"idInvoiceGroupBrands": {
			"isMongo": true
		}
	},
	{
		"idBrands": {
			"isMongo": true
		}
	},
	{
		"idProducts": {
			"isMongo": true
		}
	},
	{
		"idCart": {
			"isMongo": true
		}
	}
]
