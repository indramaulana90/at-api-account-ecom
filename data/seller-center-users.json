[
  { "name": "TRS", "username": "admin.trs@sinbad.co.id", "password": "sinbad" },
  { "name": "Unipro", "username": "admin.unipro@sinbad.co.id", "password": "sinbad" },
  { "name": "Quality", "username": "quality.sinbad@gmail.com", "password": "quality3607" },
  { "name": "Preprod", "username": "admin.techquality@sinbad.co.id", "password": "sinbad" }
]
