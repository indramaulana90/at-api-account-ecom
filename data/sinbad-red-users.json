[
    { "name": "red1", "mobilePhone": "082297872220"},
    { "name": "Scimitar-SR", "mobilePhone": "081239999999", "email": "ScimitarSR@ATAPIqe.com" },
    { "name": "Kanzun-SR", "mobilePhone": "081238888888", "email": "KanzunSR@ATAPIqe.com" },
    { "name": "Voyager-SR", "mobilePhone": "081237777777", "email": "VoyagerSR@ATAPIqe.com" },
    { "name": "Dinar-SR", "mobilePhone": "081236666666", "email": "DinarSR@ATAPIqe.com" },
    { "name": "Valkyrie-SR", "mobilePhone": "08786666666", "email": "ValkyrieSR@ATAPIqe.com" },
    { "name": "Valkyrie-Stg", "mobilePhone": "08783333333", "email": "ValkyrieStg@ATAPIqe.com" },
    { "name": "Pyramid-SR", "mobilePhone": "081234444444", "email": "PyramidSR@ATAPIqe.com" },
    { "name": "DavyJones-SR", "mobilePhone": "081232222222", "email": "DavyJonesSR@ATAPIqe.com" },
    { "name": "Chimera-SR", "mobilePhone": "081231111111", "email": "ChimeraSR@ATAPIqe.com" },
    { "name": "No Name", "mobilePhone": "085717216917", "email": "chimera@mail.com" }
  ]
  