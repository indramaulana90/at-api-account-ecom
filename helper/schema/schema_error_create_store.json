{
    "type": "object",
    "properties": {
      "status": {
        "type": "integer"
      },
      "message": {
        "type": "string"
      },
      "errorMessage": {
        "type": "string"
      },
      "type": {
        "type": "string"
      },
      "code": {
        "type": "integer"
      }
    },
    "additionalProperties": false,
    "required": [
      "status",
      "message",
      "errorMessage",
      "type",
      "code"
    ]
  }