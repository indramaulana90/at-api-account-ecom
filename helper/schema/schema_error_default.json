{
  "type": "object",
  "properties": {
      "message": {
          "type": "string"
      },
      "errorMessage": {
          "type": ["string", "object"]
      },
      "type": {
          "type": "string"
      },
      "code": {
          "type": "integer"
      }
  },
  "additionalProperties": false,
  "required": [
      "message",
      "errorMessage",
      "type",
      "code"
  ]
}