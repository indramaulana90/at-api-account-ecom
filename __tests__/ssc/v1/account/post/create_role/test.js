/**
* Author      : Farhad (farhad@sinbad.co.id)
* Modifier    :
* Domain      : account
* Description : This Main File for Testing on API post_create_role
*/

import getDataFromPool from '@at-api-core/get-data-from-pool';
import * as er from '@at-api-core/expect-response';
import endpoint from './endpoint';
import * as data from './data';
import tc from './tc';
import schema from './schema.json';
import schemaError from '../../../../../../helper/schema/schema_error_default.json';
import rs from '../../../../../../helper/responseStatusLibrary.json';
// import * as models__ from '../../../../../data/model/';
// import * as seeders__ from '../../../../../data/seeders/';

const { cookies } = getDataFromPool('ng','seller-center', 'TRS_admin.trs@sinbad.co.id');

// describe(`${tc.type[1]} ${tc.describe}`, () => {
//   test(`${tc.positive.oneCapability}`, async () => {
//     const payload = data.body();

//     resp = await endpoint(cookies, payload);
//     er.expectStatus(resp, rs.statusOk);
//     er.expectSchema(resp, schema);

//     const objectFiltered = resp.body.data[0];
//     const arrDB = await /* insert seeding data select */(objectFiltered.id);
//     const objectDB = (arrDB)[0];
//     er.expectEqualDB(resp, objectFiltered, objectDB, data.compareDB);
//   });
//   test(`${tc.positive.multipleCapabilities}`, async () => {
//     const payload = data.body();

//     resp = await endpoint(cookies, payload);
//     er.expectStatus(resp, rs.statusOk);
//     er.expectSchema(resp, schema);
//   });
// });

describe(`${tc.type[0]} ${tc.describe}`, () => {
  test(`${tc.negative.roleNameAlreadyExists}`, async () => {
    const payload = data.body();
    payload.name = 'Super Admin';

    resp = await endpoint(cookies, payload);
    er.expectStatus(resp, rs.statusBadRequest);
    er.expectSchema(resp, schemaError);
    er.expectSnapshot(resp);
  });
  test(`${tc.negative.roleNameLessThan3}`, async () => {
    const payload = data.body();
    payload.name = 'Su';

    resp = await endpoint(cookies, payload);
    er.expectStatus(resp, rs.statusUnprocessableEntity);
    er.expectSchema(resp, schemaError);
    er.expectSnapshot(resp);
  });
  test(`${tc.negative.roleNameMoreThan50}`, async () => {
    const payload = data.body();
    payload.name = 'Super Admin Super Admin Super Admin Super Admin Super Admin';

    resp = await endpoint(cookies, payload);
    er.expectStatus(resp, rs.statusUnprocessableEntity);
    er.expectSchema(resp, schemaError);
    er.expectSnapshot(resp);
  });
  test(`${tc.negative.roleNameNull}`, async () => {
    const payload = data.body();
    payload.name = null;

    resp = await endpoint(cookies, payload);
    er.expectStatus(resp, rs.statusUnprocessableEntity);
    er.expectSchema(resp, schemaError);
    er.expectSnapshot(resp);
  });
  test(`${tc.negative.roleNameEmptyString}`, async () => {
    const payload = data.body();
    payload.name = '';

    resp = await endpoint(cookies, payload);
    er.expectStatus(resp, rs.statusUnprocessableEntity);
    er.expectSchema(resp, schemaError);
    er.expectSnapshot(resp);
  });
  test(`${tc.negative.roleNameNotSet}`, async () => {
    const payload = data.body();
    Reflect.deleteProperty(payload, 'name');

    resp = await endpoint(cookies, payload);
    er.expectStatus(resp, rs.statusUnprocessableEntity);
    er.expectSchema(resp, schemaError);
    er.expectSnapshot(resp);
  });
  test(`${tc.negative.descLessThan5}`, async () => {
    const payload = data.body();
    payload.description = 'test';

    resp = await endpoint(cookies, payload);
    er.expectStatus(resp, rs.statusUnprocessableEntity);
    er.expectSchema(resp, schemaError);
    er.expectSnapshot(resp);
  });
  test(`${tc.negative.descMoreThan1000}`, async () => {
    const payload = data.body();
    payload.description = 'test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test test';

    resp = await endpoint(cookies, payload);
    er.expectStatus(resp, rs.statusUnprocessableEntity);
    er.expectSchema(resp, schemaError);
    er.expectSnapshot(resp);
  });
  test(`${tc.negative.descNull}`, async () => {
    const payload = data.body();
    payload.description = null;

    resp = await endpoint(cookies, payload);
    er.expectStatus(resp, rs.statusUnprocessableEntity);
    er.expectSchema(resp, schemaError);
    er.expectSnapshot(resp);
  });
  test(`${tc.negative.descEmptyString}`, async () => {
    const payload = data.body();
    payload.description = '';

    resp = await endpoint(cookies, payload);
    er.expectStatus(resp, rs.statusUnprocessableEntity);
    er.expectSchema(resp, schemaError);
    er.expectSnapshot(resp);
  });
  test(`${tc.negative.descNotSet}`, async () => {
    const payload = data.body();
    Reflect.deleteProperty(payload, 'description');

    resp = await endpoint(cookies, payload);
    er.expectStatus(resp, rs.statusUnprocessableEntity);
    er.expectSchema(resp, schemaError);
    er.expectSnapshot(resp);
  });

  test(`${tc.negative.capabilitiesNull}`, async () => {
    const payload = data.body();
    payload.capabilities = null;

    resp = await endpoint(cookies, payload);
    er.expectStatus(resp, rs.statusUnprocessableEntity);
    er.expectSchema(resp, schemaError);
    er.expectSnapshot(resp);
  });
  test(`${tc.negative.capabilitiesEmptyString}`, async () => {
    const payload = data.body();
    payload.capabilities = '';

    resp = await endpoint(cookies, payload);
    er.expectStatus(resp, rs.statusUnprocessableEntity);
    er.expectSchema(resp, schemaError);
    er.expectSnapshot(resp);
  });
  test(`${tc.negative.capabilitiesNotSet}`, async () => {
    const payload = data.body();
    Reflect.deleteProperty(payload, 'capabilities');

    resp = await endpoint(cookies, payload);
    er.expectStatus(resp, rs.statusUnprocessableEntity);
    er.expectSchema(resp, schemaError);
    er.expectSnapshot(resp);
  });
});

describe(`${tc.type[0]} ${tc.describe}`, () => {
  test(`${tc.negative.xPlatformInvalid}`, async () => {
    const payload = data.body();

    resp = await endpoint(cookies, payload).set('x-platform', 'error');
    er.expectStatus(resp, rs.statusUnauthorized);
    er.expectUnauthorized(resp);
  });

  test(`${tc.negative.xPlatformNotSet}`, async () => {
    const payload = data.body();

    resp = await endpoint(cookies, payload).unset('x-platform');
    er.expectStatus(resp, rs.statusUnauthorized);
    er.expectUnauthorized(resp);
  });

  data.loopingErrorCookies.forEach(async (element) => {
    test(`${element.title}`, async () => {
      const payload = data.body();
      const cookiesNegative = element.value;

      resp = await endpoint(cookiesNegative, payload);
      er.expectStatus(resp, rs.statusUnauthorized);
      er.expectUnauthorized(resp);
    });
  });
});
