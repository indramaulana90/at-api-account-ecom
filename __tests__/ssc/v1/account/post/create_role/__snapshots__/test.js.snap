// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@negative post create_role -- As a Supplier Admin, I should not be able to create new role because capabilities is empty string 1`] = `
Object {
  "code": 40020000001,
  "errorMessage": Object {
    "capabilities": "capabilities should not be empty",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative post create_role -- As a Supplier Admin, I should not be able to create new role because capabilities is not set 1`] = `
Object {
  "code": 40020000001,
  "errorMessage": Object {
    "capabilities": "capabilities should not be empty",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative post create_role -- As a Supplier Admin, I should not be able to create new role because capabilities is null 1`] = `
Object {
  "code": 40020000001,
  "errorMessage": Object {
    "capabilities": "capabilities should not be empty",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative post create_role -- As a Supplier Admin, I should not be able to create new role because desc is empty string 1`] = `
Object {
  "code": 40020000001,
  "errorMessage": Object {
    "description": "description should not be empty",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative post create_role -- As a Supplier Admin, I should not be able to create new role because desc is less than 5 chars 1`] = `
Object {
  "code": 40020000001,
  "errorMessage": Object {
    "description": "description must be longer than or equal to 5 characters",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative post create_role -- As a Supplier Admin, I should not be able to create new role because desc is more than 1000 chars 1`] = `
Object {
  "code": 40020000001,
  "errorMessage": Object {
    "description": "description must be shorter than or equal to 1000 characters",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative post create_role -- As a Supplier Admin, I should not be able to create new role because desc is not set 1`] = `
Object {
  "code": 40020000001,
  "errorMessage": Object {
    "description": "description should not be empty",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative post create_role -- As a Supplier Admin, I should not be able to create new role because desc is null 1`] = `
Object {
  "code": 40020000001,
  "errorMessage": Object {
    "description": "description should not be empty",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative post create_role -- As a Supplier Admin, I should not be able to create new role because role name is already exists 1`] = `
Object {
  "code": 40020000341,
  "errorMessage": "Role is exists",
  "message": "Please choose another role",
  "type": "BAD_REQUEST",
}
`;

exports[`@negative post create_role -- As a Supplier Admin, I should not be able to create new role because role name is empty string 1`] = `
Object {
  "code": 40020000001,
  "errorMessage": Object {
    "name": "name should not be empty",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative post create_role -- As a Supplier Admin, I should not be able to create new role because role name is less than 3 chars 1`] = `
Object {
  "code": 40020000001,
  "errorMessage": Object {
    "name": "name must be longer than or equal to 3 characters",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative post create_role -- As a Supplier Admin, I should not be able to create new role because role name is more than 50 chars 1`] = `
Object {
  "code": 40020000001,
  "errorMessage": Object {
    "name": "name must be shorter than or equal to 50 characters",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative post create_role -- As a Supplier Admin, I should not be able to create new role because role name is not set 1`] = `
Object {
  "code": 40020000001,
  "errorMessage": Object {
    "name": "name should not be empty",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative post create_role -- As a Supplier Admin, I should not be able to create new role because role name is null 1`] = `
Object {
  "code": 40020000001,
  "errorMessage": Object {
    "name": "name should not be empty",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;
